crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -357,17 +357,30 @@ impl Format<PyFormatContext<'_>> for FormatTrailingEndOfLineComment<'_> {
 
         let normalized_comment = normalize_comment(self.comment, source)?;
 
-        // Start with 2 because of the two leading spaces.
-        let mut reserved_width = 2;
-
-        // SAFE: The formatted file is <= 4GB, and each comment should as well.
-        #[allow(clippy::cast_possible_truncation)]
-        for c in normalized_comment.chars() {
-            reserved_width += match c {
-                '\t' => f.options().tab_width().value(),
-                c => c.width().unwrap_or(0) as u32,
+        // Trim the normalized comment to detect excluded pragmas (don't handle NBSP).
+        let trimmed = strip_comment_prefix(&normalized_comment)?.trim_start();
+
+        // Don't reserve width for excluded pragma comments.
+        let reserved_width = if ["noqa", "type:", "pyright:", "pylint"]
```

Comment:
😍

Reply:
To make sure I'm clear, your solution works since we're intentionally disregarding the *bare* `# noqa` lints `ruff` no longer expects to support?

```
warning: Invalid `# noqa` directive on scratch.py:16: expected `:` followed by a comma-separated list of codes (e.g., `# noqa: F401, F841`).
```

Reply:
Hmm good point, we may need the `starts_width` in the case the colon is missing to cover ruff's `noqa` :(

Reply:
fccdcfe2550c3fdb40bc73bb3e974ae3d1b8958f